### PR TITLE
Safe part 1

### PR DESCRIPTION
**Pull request by tomquirk**
## What does this PR do and why?

- Loads queued transactions from Gnosis and links to them 
- Provides framework for adding expanding sections for the previews of specific transactions
- Test at mainnet: `v2/p/233/safe`

https://user-images.githubusercontent.com/96150256/190039705-cef0bb8a-527b-4f3c-83a4-136776a377b0.mp4

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

File: src/components/v2/V2Project/V2ProjectSafeDashboard/SafeTransaction.tsx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+const ReconfigureFundingCyclesOfTransaction = ({
+  transaction,
+}: SafeTransactionComponentProps) => {
+  // const [expanded, setExpanded] = useState<boolean>(false)
+
+  const { contracts } = useContext(V2UserContext)
+
+  if (!contracts || !transaction.data) return null
+
+  // const decodedData = contracts['JBController'].interface.parseTransaction({
```
Comment (johnnyd-eth) [inline src/components/v2/V2Project/V2ProjectSafeDashboard/SafeTransaction.tsx:54]: Will leave this here for upcoming PR

**Comment by johnnyd-eth** [comment]
@tomquirk this is ready for a look! The only issue I'm seeing in the test case is that the Safe API seems to be sending back some already-rejected tx's, even though we're passing `queued: true` to the request. This seems to be an error on their part. 

I dug a bit but a solution wasn't obvious, so I think we chase the up in a follow-up.

File: src/components/v2/V2Project/V2ProjectSafeDashboard/SafeTransaction.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+export function SafeTransaction({
+  transaction,
+}: {
+  transaction: SafeTransactionType
+}) {
+  const {
+    theme: { colors },
+  } = useContext(ThemeContext)
+  const [hover, setHover] = useState<boolean>(false)
```
Comment (tomquirk) [inline src/components/v2/V2Project/V2ProjectSafeDashboard/SafeTransaction.tsx:98]: **suggestion** shouldn't ever set hover state with JS.

I like what you've done with the styling though. It looks like we already have a class for this anyway 💪 

`.hover-bg-l2`

File: src/components/v2/V2Project/V2ProjectSafeDashboard/SafeTransaction.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+  const TransactionContent = useMemo(() => {
+    if (!method) return GenericSafeTransaction // only the header
+    return TRANSACTION_METHOD_COMPONENTS_MAP[method] ?? GenericSafeTransaction
+  }, [method])
+
+  if (!method) return null
+
+  return (
+    <ExternalLink
```
Comment (tomquirk) [inline src/components/v2/V2Project/V2ProjectSafeDashboard/SafeTransaction.tsx:123]: Oh nice, i like it 👍

File: src/styles/utilities.scss [hunk 4]
```diff
@@ -30,10 +30,18 @@
...
 
+.hover-box-shadow, .hover-box-shadow-sm {
+  transition: box-shadow 0.3s;
+}
+
 .hover-box-shadow:hover {
   box-shadow: 3px 0px 11px 0px var(--boxShadow-primary)
 }
 
+.hover-box-shadow-sm:hover {
```
Comment (tomquirk) [inline src/styles/utilities.scss:41]: big moves. love it

File: src/utils/safe.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import { SafeTransactionType } from 'components/v2/V2Project/V2ProjectSafeDashboard/V2ProjectSafeDashboard'
+
+// Links to a Gnosis safe multisig
+export const generateSafeUrl = (address: string) =>
+  `https://gnosis-safe.io/app/eth:${address}/transactions/queue`
+
+// Links to specific transaction on Gnosis Safe app
+export const generateSafeTxUrl = (transaction: SafeTransactionType) =>
```
Comment (tomquirk) [inline src/utils/safe.ts:8]: **praise** beautiful

File: src/utils/safe.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import { SafeTransactionType } from 'components/v2/V2Project/V2ProjectSafeDashboard/V2ProjectSafeDashboard'
```
Comment (tomquirk) [inline src/utils/safe.ts:1]: **suggestion** WDYT about moving this file to `lib/`?

I think lib is a better spot for service-specific helpers (like safe, snapshot etc.)

File: src/components/v2/V2Project/V2ProjectSafeDashboard/V2ProjectSafeDashboard.tsx [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+      {isLoading && <div style={{ marginTop: 20 }}>Loading...</div>}
+
+      {!isLoading && (
+        <Tabs
+          style={{ marginTop: 20 }}
+          tabBarStyle={{ border: 'unset', marginBottom: 15 }}
+        >
+          <TabPane tab={<Trans>Queued</Trans>} key="1">
+            {queuedSafeTransactions?.results.map(
```
Comment (tomquirk) [inline src/components/v2/V2Project/V2ProjectSafeDashboard/V2ProjectSafeDashboard.tsx:77]: **question** Is the transaction state (e.g. if it's rejected) in the transaction data? If it is, maybe we could filter them out of this array.

Maybe (untested, just a guess):

```
queuedSafeTransactions.filter(tx => !tx.isExecuted)...
```

**Review by tomquirk — COMMENTED** [review]
@johnnyd-eth looks so great! Thanks heaps for picking up the slack on this. Real exciting.

I left a few nitpicks. One other thing: can we make the tabs look the same as the Projects page?

Screen Shot 2022-09-14 at 9 01 40 PM

File: src/utils/safe.ts (hunk 6 above)
Comment (johnnyd-eth) [reply src/utils/safe.ts:1]: Good call

**Comment by johnnyd-eth** [comment]
@tomquirk latest commit:
1. Creates new `Tab` component to use in for the safe tabs and `projects` page tabs.
2. Filters queued safe transactions by `safeTxGas` (removes any that aren't 0). As far as I can see, executed or rejected transactions always have some gas, and queued transactions are always 0. 

- I have tested some extra safe tx's and some such as [this](https://gnosis-safe.io/app/eth:0x07f356Fdb3EA88Fad8607A58434b35F528Ce9b1b/transactions/multisig_0x07f356Fdb3EA88Fad8607A58434b35F528Ce9b1b_0x348bad414c42f91c512c893cab6b73b79975c38ab8402ff09f980849aa566f32) are coming through with null data. 
- I'm not showing these. 
- I've reached out to Safe about what we can do about these. Think we should just merge what we've got for now.

**Comment by tomquirk** [comment]
@johnnyd-eth LFG!!! I made one last commit, check it out if you're interested: https://github.com/jbx-protocol/juice-interface/pull/1965/commits/57ce2de00e43970ab7121bfa184b6cf72e943d47

Taking note of a couple of UX things that we should implement later:
- this list should have an initial limit of, say 10 txs. Then we have a **load more** button at the bottom of the list
- ... i think that's it actually.

Merging 🎉 🎉 🎉 🎉 

---

Note: One thing I've learnt recently is to generally avoid `closure` components; components that are defined inside other components. Can lead to buggy stuff with re-rendering. I moved ProjectTab outside of the ProjectTabs scope, for example.

**Pull request merged**